### PR TITLE
removal of unnecssary parsing if httpMethod is null

### PR DESCRIPTION
**Pull request by guptasourabh**
If httpMethod is null then we don't need to parse and set the other values

### CODE REVIEW

File: retrofit/src/main/java/retrofit2/ServiceMethod.java [hunk 1]
```diff
@@ -282,6 +282,8 @@ private void parseHttpMethodAndPath(String httpMethod, String value, boolean has
         throw methodError("Only one HTTP method is allowed. Found: %s and %s.",
             this.httpMethod, httpMethod);
       }
+      if (null == httpMethod) return;
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit2/ServiceMethod.java]: This will never be true.

Comment (guptasourabh) [inline retrofit/src/main/java/retrofit2/ServiceMethod.java]: if we go through the code in the ServiceMethod build after parseMethodAnnotations , we are checking if httpMethod is null or not and then throwing the exception , so if we check at parseHttpMethodPath itself , we don't need to do value parsing.

**Comment by JakeWharton** [comment]
The exception checks that the method isn't called twice. The value of
httpMethod is guaranteed to never be null by Java since it comes from an
annotation.

**Comment by guptasourabh** [comment]
@JakeWharton requested changes done

**Comment by guptasourabh** [comment]
Moved it into new PR https://github.com/square/retrofit/pull/2219

**Pull request closed without merging**